src/main/java/io/fabric8/launcher/booster/catalog/Booster.java

```diff
@@ -257,7 +257,10 @@ public Booster forEnvironment(String environmentName) {
      */
     @SuppressWarnings("unchecked")
     public Map<String, Object> getMetadata() {
-        return data.containsKey(KEY_METADATA) ? Collections.unmodifiableMap((Map<String, Object>) data.get(KEY_METADATA)) : Collections.emptyMap();
+        Map<String, Object> metadata = (Map<String, Object>) data.get(KEY_METADATA);
```

Comment:
I don't get why this change is necessary. How can `get(x)` return `null` if `containsKey(x)` was `true`? Is someone somewhere doing a `.put(x, null)`? If so that's really what should be fixed I think.

Reply:
The YAML parser automatically puts `null` in the `metadata` key when if finds a `booster.yaml` like this: https://github.com/fabric8-launcher/launcher-booster-catalog/blob/v39/spring-boot/current-redhat/health-check/booster.yaml#L7

Reply:
If you run `BoosterTest`, you will see the NPE happening

Reply:
It seems that `BoosterTest` is conflicting with other tests. I don't exactly know why, but I'm working on it